pytorch_lightning/trainer/progress.py

```diff
@@ -141,19 +142,21 @@ def scheduler_steps(self) -> int:
 
 
 @dataclass
-class TrainingProgress(Progress):
-    """
-    Extends ``Progress`` with training specific attributes
+class TrainingLoopProgress(LoopProgress):
 
-    Args:
-        optimization: Tracks optimization progress
-    """
-    optimization: OptimizationProgress = field(default_factory=OptimizationProgress)
+    optimizer_idx: Optional[int] = field(default_factory=int)
```

Comment:
```suggestion
    optimizer_idx: int = 0
```

Reply:
I would prefer to keep optional, as `manual_optimization` don't have an `optimizer_idx`. None seems more appropriate.

Reply:
Sure, then

```suggestion
    optimizer_idx: Optional[int] = None
```